services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -0,0 +1,180 @@
+                ["a-switch", 0, getTopology().islsForActiveSwitches.find {
+                    it.aswitch?.inPort && it.aswitch?.outPort
+                }]
+        ]
+    }
+
+    @Tags(SMOKE)
+    def "Port history should not be returned in case timeline is incorrect (timeBefore > timeAfter)"() {
+        given: "A direct link"
+        def timestampBefore = System.currentTimeSeconds()
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+
+        when: "Execute port DOWN on the src switch"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LINK_DOWN"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        and: "Execute port UP on the src switch"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LIVE"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Port history is created"
+        Long timestampAfter = System.currentTimeSeconds()
+        def portHistory = northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfter)
+        assert portHistory.size() == 2
+
+        when: "Get port history for incorrect timeline"
+        def portH = northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampAfter, timestampBefore)
+
+        then: "Port history is NOT returned"
+        portH.isEmpty()
+    }
+
+    def "Port history should not be returned in case port/switch were never exist"() {
+        when: "Try to get port history for incorrect port and switch"
+        def portHistory = northboundV2.getPortHistory(NON_EXISTENT_SWITCH_ID, 99999)
+
+        then: "Port history is empty"
+        portHistory.isEmpty()
+    }
+
+    def "Port history is available when switch is DEACTIVATED"() {
+        given: "A direct link"
+        def timestampBefore = System.currentTimeSeconds()
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+
+        when: "Execute port DOWN on the src switch"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LINK_DOWN"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        and: "Execute port UP on the src switch"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LIVE"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Port history is created"
```

Comment:
already tested in previous tests and is not the goal of this test

Reply:
needless steps were removed